components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/dao/impl/AuthenticatorManagementFacade.java

```diff
@@ -0,0 +1,218 @@
+                        tenantId);
+                validateAuthenticatorProperties(newAuthenticatorConfig);
+                return endpointConfigManager.resolveEndpointConfigurations(
+                            dao.updateUserDefinedLocalAuthenticator(existingAuthenticatorConfig, newAuthenticatorConfig,
+                                    tenantId), tenantId);
+            });
+        } catch (TransactionException e) {
+            throw handleAuthenticatorMgtException(AuthenticatorMgtError.ERROR_WHILE_UPDATING_AUTHENTICATOR, e,
+                    newAuthenticatorConfig.getName());
+        }
+    }
+
+    /**
+     * Get user defined local authenticator by name and resolving associated data (endpoint configurations)
+     * by invoking external service.
+     *
+     * @param authenticatorConfigName Name of the user defined local authenticator.
+     * @param tenantId                Tenant ID.
+     * @return User defined local authenticator.
+     * @throws AuthenticatorMgtException If an error occurs while retrieving the user defined local authenticator.
+     */
+    @Override
+    public UserDefinedLocalAuthenticatorConfig getUserDefinedLocalAuthenticator(
+            String authenticatorConfigName, int tenantId) throws AuthenticatorMgtException {
+
+        NamedJdbcTemplate jdbcTemplate = new NamedJdbcTemplate(IdentityDatabaseUtil.getDataSource());
+        try {
+            return jdbcTemplate.withTransaction(template ->
+                    endpointConfigManager.resolveEndpointConfigurations(dao.getUserDefinedLocalAuthenticator(
+                            authenticatorConfigName, tenantId), tenantId));
+        } catch (TransactionException e) {
+            throw handleAuthenticatorMgtException(AuthenticatorMgtError.ERROR_WHILE_RETRIEVING_AUTHENTICATOR_BY_NAME, e,
+                    authenticatorConfigName);
+        }
+    }
+
+    /**
+     * Get all user defined local authenticators and resolving associated data (endpoint configurations)
+     * by invoking external service.
+     *
+     * @param tenantId Tenant ID.
+     * @return List of user defined local authenticators.
+     * @throws AuthenticatorMgtException If an error occurs while retrieving all user defined local authenticators.
+     */
+    @Override
+    public List<UserDefinedLocalAuthenticatorConfig> getAllUserDefinedLocalAuthenticators(int tenantId)
+            throws AuthenticatorMgtException {
+
+        NamedJdbcTemplate jdbcTemplate = new NamedJdbcTemplate(IdentityDatabaseUtil.getDataSource());
+        try {
+            return jdbcTemplate.withTransaction(template -> {
+                List<UserDefinedLocalAuthenticatorConfig> configList =
+                        dao.getAllUserDefinedLocalAuthenticators(tenantId);
+                for (UserDefinedLocalAuthenticatorConfig config : configList) {
+                    endpointConfigManager.resolveEndpointConfigurations(config, tenantId);
+                }
```

Comment:
Ideally we should have a bulk operation supported from actions side. Shall we add a note of this similar to the way this was tracked in IdP Management